File: 백준/Bronze/1978. 소수 찾기/소수 찾기.py
import sys
import math
n = int(sys.stdin.readline())
num = list(map(int, sys.stdin.readline().split()))
cnt = 0

for i in num:
    if i < 2:    # 1은 소수 X.
        continue
    else:
        for j in range(2, int(math.sqrt(i)) + 1):   # int()는 소숫점 버림으로써 정수형으로 형변환.
            if i % j == 0:
                break
        else:    # break 등으로 중간에 빠져나오지 않고 for문 전부 순회했을 때 수행
            cnt += 1
print(cnt)

File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.js
// 특정 범위 안에 존재하는 모든 소수 찾을 때는 '에라토스테네스의 체' 이용
function solution(n) {
    const isPrime = new Array(n + 1).fill(true);
    isPrime[0] = false;
    isPrime[1] = false;
    
    for (let i = 2; i <= Math.sqrt(n); i++) {
        if (isPrime[i] === true) {
            for (let j = i + i; j < isPrime.length; j += i) {
                isPrime[j] = false;
            }
        }
    }
    
    return isPrime.filter(num => num === true).length;
}